specifyweb/frontend/js_src/lib/components/QueryBuilder/Fields.tsx

```diff
@@ -61,9 +62,71 @@ export function QueryFields({
     | ((line: number, direction: 'down' | 'up') => void)
     | undefined;
   readonly onOpenMap: ((line: number) => void) | undefined;
+  readonly onChangeFields?: ((fields: RA<QueryField>) => void) | undefined;
 }): JSX.Element {
   const fieldsContainerRef = React.useRef<HTMLUListElement | null>(null);
 
+  const [draggedItem, setDraggedItem] = React.useState<QueryField | null>(null);
+
+  const [isDragging, setIsDragging] = React.useState(false);
+
+  React.useEffect(() => {
+    if (isDragging) {
+      const listNode = fieldsContainerRef.current;
+      if (listNode === null) return;
+      const { top, bottom } = listNode.getBoundingClientRect();
+      const windowHeight = window.innerHeight;
+
+      if (top < 0) {
+        // The dragged item is at the top of the screen
+        listNode.scrollTop -= Math.abs(top);
+      } else if (bottom > windowHeight) {
+        // The dragged item is at the bottom of the screen
+        listNode.scrollTop += bottom - windowHeight;
+      }
+    }
+  }, [isDragging]);
+
+  const handleDragStart = (
+    event: React.DragEvent<HTMLDivElement>,
+    index: number
+  ) => {
+    setDraggedItem(fields[index]);
```

Comment:
I am assuming a lot of the code in this component is based on online/examples. If so, can you provide links to those places as comments before the code?

i.e, I often put in the code something like

```js
/**
 * Inspired by:
 * <link>
 */
```

```js
/**
 * Based on:
 * <link>
 */
```

or just docs that you think would be helpful for someone looking at this in the future